rust_dev_preview/lambda/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_config::{meta::region::RegionProviderChain, SdkConfig};
+use aws_sdk_lambda::{Client, Region, PKG_VERSION};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Opt {
+    /// The AWS Region.
+    #[structopt(short, long)]
+    pub region: Option<String>,
+
+    /// Whether to display additional runtime information.
+    #[structopt(short, long)]
+    pub verbose: bool,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct ArnOpt {
+    #[structopt(flatten)]
+    pub base: Opt,
+
+    /// The AWS Lambda function's Amazon Resource Name (ARN).
+    #[structopt(short, long)]
+    pub arn: String,
+}
+
+pub fn make_region_provider(opt: Option<String>) -> RegionProviderChain {
+    RegionProviderChain::first_try(opt.map(Region::new))
+        .or_default_provider()
+        .or_else(Region::new("us-west-2"))
+}
+
+pub async fn make_client(opt: Opt) -> (Client, SdkConfig) {
```

Comment:
I think it would be clearer if this was renamed to something like `configure_sdk` and made to only return a `SdkConfig` with the arguments applied. The `Client` creation from the shared config is a pretty trivial piece.

Reply:
I had done that... in this other commit that I forgot to push XD